static/src/javascripts/projects/common/utils/steady-page.js

```diff
@@ -0,0 +1,115 @@
+/*
+ Utility: steady-insert.js
+ Insert components into web-pages without jumping them around by
+ moving the current scroll position by the same distance as the height of the
+ inserted element.
+ Note: Inserted elements must have a fixed height when rendered
+ (i.e: no changing the height by loading an image, lazy-loading content etc)
+ */
+
+define([
+    'common/utils/fastdom-promise'
+], function (
+    fastdom
+) {
+    var currentFrameContainers = [];
+    var currentScrollPos;
+    var heightToAdd = 0;
+    var options = {
+        pixelGrace: 100 // Give the scroll some distance as it's likely the user has scrolled past the top of the screen
+    };
+
+    /*
+     * Height with margins
+     */
+    function outerHeight(el) {
+        var height = el.offsetHeight;
```

Comment:
Agreed with @sndrs - this would require you to know that this lib reads the DOM.